Add description of this patch here so that it can be sent upstream.

Signed-off-by: [g2] add name here

Index: linux-2.6.19-rc2/drivers/mtd/redboot.c
===================================================================
--- linux-2.6.19-rc2.orig/drivers/mtd/redboot.c	2006-10-19 00:21:11.000000000 -0700
+++ linux-2.6.19-rc2/drivers/mtd/redboot.c	2006-10-19 00:21:27.000000000 -0700
@@ -85,6 +85,10 @@
 
 	numslots = (master->erasesize / sizeof(struct fis_image_desc));
 	for (i = 0; i < numslots; i++) {
+		if (buf[i].name[0] == 0xff) {
+			i = numslots;
+			break;
+		}
 		if (!memcmp(buf[i].name, "FIS directory", 14)) {
 			/* This is apparently the FIS directory entry for the
 			 * FIS directory itself.  To discover whether the entries
@@ -155,7 +159,7 @@
 		struct fis_list *new_fl, **prev;
 
 		if (buf[i].name[0] == 0xff)
-			continue;
+			break;
 		if (!redboot_checksum(&buf[i]))
 			break;
 
